File: algorithms/1941.check-if-all-characters-have-equal-number-of-occurrences/soluiton.go
func areOccurrencesEqual(s string) bool {
	count := map[rune]int{}

	for _, r := range s {
		if _, ok := count[r]; ok {
			count[r] += 1
		} else {
			count[r] = 1
		}
	}

	n := 0
	for _, v := range count {
		if n > 0 {
			if n != v {
				return false
			}
		} else {
			n = v
		}
	}

	return true
}

File: algorithms/1941.check-if-all-characters-have-equal-number-of-occurrences/soluiton.rs
use std::collections::{HashMap, HashSet};

impl Solution {
    pub fn are_occurrences_equal(s: String) -> bool {
        let mut count: HashMap<char, i32> = HashMap::new();
        for c in s.chars() {
            *count.entry(c).or_insert(0) += 1;
        }

        count
            .into_iter()
            .map(|(_, n)| n)
            .collect::<HashSet<i32>>()
            .len()
            == 1
    }
}
